### PR TITLE
SCC: make vertical dimension optional

### PR DESCRIPTION
**Pull request by awnawab**
This PR makes the vertical dimension optional for the SCC family of transforms.

NB: this is stacked on top of #268.

### CODE REVIEW

File: transformations/transformations/single_column_coalesced.py [hunk 1]
```diff
@@ -362,7 +362,8 @@ def kernel_annotate_vector_loops_openacc(cls, routine, horizontal, vertical):
         argument_map = CaseInsensitiveDict({a.name: a for a in routine.arguments})
         private_arrays = [v for v in routine.variables if not v.name in argument_map]
         private_arrays = [v for v in private_arrays if isinstance(v, sym.Array)]
-        private_arrays = [v for v in private_arrays if not any(vertical.size in d for d in v.shape)]
+        if vertical:
```
[the remark this answers is not in the document]
Comment (awnawab) [reply transformations/transformations/single_column_coalesced.py:365]: Yes that's a good idea. In fact, now that I think about it, only statically declared arrays should be marked for privatisation. This is the logic that we follow in the stack and hoist variants, so any temporary array that isn't  statically declared will be either allocated on the stack or hoisted, and thus we shouldn't mark it for privatisation. Let me know please if you agree and I'll make the change.

File: transformations/tests/test_single_column_coalesced.py [hunk 2]
```diff
@@ -808,7 +810,7 @@ def test_single_column_coalesced_hoist_openacc(frontend, horizontal, vertical, b
     synthesis.apply(kernel, role='kernel', item=kernel_item, successors=(module_item,))
 
     annotate = SCCAnnotateTransformation(
```
[the remark this answers is not in the document]
Comment (awnawab) [reply transformations/tests/test_single_column_coalesced.py:812]: This was perhaps a lazy way to show that the way we use the vertical dimension in the scc annotate transformation really has no effect. `nproma, nlev` arrays can't be demoted and shouldn't be privatised. `nlev` only arrays also shouldn't be privatised and will in fact be hoisted if requested. I am now thinking I should have just removed `vertical` entirely from the `SCCAnnotateTransformation`.

File: transformations/transformations/single_column_coalesced.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (reuterbal) [reply transformations/transformations/single_column_coalesced.py:365]: That sounds sensible, yes.

**Comment by reuterbal** [comment]
Thanks. This has accumulated some conflicts and needs a rebase but I suggest to take care of the changes requested in #268 first and then address @MichaelSt98's comment.

**Review by reuterbal — APPROVED** [review]
Thanks, very nice!

I can confirm this works correctly with CLOUDSC and performance is unaffected even for hoisting after removing the vertical dimension from the Loki config file.

**Comment by mlange05** [comment]
Confirmed clean with ec-physics regression.

**Pull request merged**